File: Scadex.Model/Dtos/CameraCaptureSetting/Queries/CameraCaptureSettingDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.CameraCaptureSetting.Queries;

public class CameraCaptureSettingDto : IDto
{
    public int SnapshotTimeoutMs { get; set; }
    public int SnapshotCacheSeconds { get; set; }
    public string CaptureRoot { get; set; } = null!;
    public int CaptureRetentionDays { get; set; }
    public int MaxClipDurationSec { get; set; }
    public int ClipFinalizeGraceMs { get; set; }
}

File: Scadex.WebUI/src/models/cameraCaptureSetting/queries/cameraCaptureSettingDto.ts
/** Ayna: Scadex.Model/Dtos/CameraCaptureSetting/Queries/CameraCaptureSettingDto.cs */

/**
 * Kamera çekim (anlık görüntü + klip) ayarları.
 *
 * Medya geçidi ayarlarıyla aynı yerde durur ama AYRI bir tablo ve ayrı bir servistir
 * (`ICameraCaptureSettingService`): ayar nesnesi başına bir servis kuralı gereği.
 * Kaydedildiği an etkilidir.
 */
export interface CameraCaptureSettingDto {
  /** Kameradan ISAPI ile anlık görüntü çekerken beklenecek süre. */
  snapshotTimeoutMs: number;
  /** Aynı kameranın anlık görüntüsünün önbellekte tutulma süresi. `0` = önbellek yok. */
  snapshotCacheSeconds: number;
  /** `wwwroot` ALTINDA göreli bir yol — mutlak yol veya `..` kabul edilmez. */
  captureRoot: string;

  /** Çekimin saklanma süresi (gün). `0` = süresiz. */
  captureRetentionDays: number;

  /** Tek bir klibin isteyebileceği en uzun süre. */
  maxClipDurationSec: number;
  /** MediaMTX'in dosyayı kapatması için klip bittikten sonra beklenen pay. */
  clipFinalizeGraceMs: number;
}
